buildsystem/generator/make.go

```diff
@@ -0,0 +1,300 @@
+package generator
+
+import (
+	"io/ioutil"
+
+	"github.com/evergreen-ci/shrub"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// Make represents an evergreen config generator for Makefile-based projects.
+type Make struct {
+	Tasks    map[string]MakeTask    `yaml:"tasks,omitempty"`
+	Variants map[string]MakeVariant `yaml:"variants"`
```

Comment:
should these be lists instead?

Reply:
I'm also, not sure about this, but it seems like it might make sense to split out generator implementation/models (as with options), but I'm not sure.

Reply:
It would be significantly more difficult for them to be slices since you would have to do lots of duplicate checking when merging file parts and wouldn't provide a great deal of use to make them slices since we're assuming there's 1 definition for each task/variant. I don't think the YAML file reads particularly differently and to me is actually prefereable if you look at the example.

We could split the generator models, but they would look identical so it would just be copy-paste.